app/services/sso/openid_connect.rb

```diff
@@ -0,0 +1,56 @@
+module SSO
+  class OpenidConnect < Base
+    delegate :session, :to => :controller
+    attr_reader :current_user
+
+    def available?
+      controller.api_request? && jwt_token_set?
+    end
+
+    def authenticate!
+      payload = jwt_token.decode
+      user = create_user_from_jwt(payload)
+      @current_user = user
+      store(payload)
+      user&.login
+    rescue JWT::DecodeError
+      Rails.logger.error "JWT SSO: Failed to decode JWT."
+      nil
+    end
+
+    def authenticated?
+      self.user = User.current.presence || authenticate!
+    end
+
+    private
+
+    def jwt_token
+      @jwt_token ||= jwt_token_from_request
+    end
+
+    def jwt_token_from_request
+      token = request.authorization.split(' ')[1]
+      OidcJwtToken.new(token)
+    end
+
+    def jwt_token_set?
+      request.authorization.present? && request.authorization.start_with?('Bearer')
+    end
+
+    def store(payload)
```

Comment:
I'm not particularly fond of this method name. I think it should emphasize that it stores data in the session. How about

```ruby
def update_session(sso_method: self.class.to_s, expires_at:)`.
  session[:sso_method] = sso_method
  session[:expires_at] = expires_at
end
```